### PR TITLE
Optimize GitHub Workflows with Caching

### PR DESCRIPTION
**Pull request by kshitijk4poor**

### CODE REVIEW

**Comment by kazet** [comment]
makes sense! would it be possible to also add caching for test Docker build? This is where the runners spend the most time

File: .github/workflows/docker_nightly.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
 
+      #Caching Docker layers
```
Comment (kazet) [inline .github/workflows/docker_nightly.yml:24]: I don't think this comment is needed

Comment (kazet) [inline .github/workflows/docker_nightly.yml:24]: I don't think this comment is needed

File: .github/workflows/liccheck.yml [hunk 2]
```diff
@@ -10,15 +11,28 @@ jobs:
     steps:
       - name: Check out repository
         uses: actions/checkout@v2
+
       - name: Set up Python 3.11
         uses: actions/setup-python@v2
         with:
           python-version: "3.11"
+
+      # Caching Python dependencies
```
Comment (kazet) [inline .github/workflows/liccheck.yml:20]: I don't think this comment is needed

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -10,13 +11,34 @@ jobs:
     steps:
       - name: Check out repository
         uses: actions/checkout@v2
+
       - name: Set up Python 3.11
         uses: actions/setup-python@v2
         with:
           python-version: "3.11"
+
+      # Caching Python dependencies
```
Comment (kazet) [inline .github/workflows/lint.yml:20]: I don't think this comment is needed

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -10,13 +11,34 @@ jobs:
...
+      - name: Cache Python dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      # Cache pre-commit hooks
```
Comment (kazet) [inline .github/workflows/lint.yml:29]: I don't think this comment is needed

Comment (kazet) [inline .github/workflows/lint.yml:29]: I don't think this comment is needed

Comment (kazet) [inline .github/workflows/lint.yml:29]: I don't think this comment is needed

File: artemis/modules/nuclei.py [hunk 5]
```diff
@@ -152,20 +152,21 @@ def run_multiple(self, tasks: List[Task]) -> None:
 
         links_per_task = {}
         links = []
-        for task in tasks:
-            links_per_task[task.uid] = self._get_links(get_target_url(task))
-            self.log.info("Links for %s: %s", get_target_url(task), links_per_task[task.uid])
-            links.extend(links_per_task[task.uid])
+        if not Config.Modules.Nuclei.NUCLEI_AUTOMATIC_SCAN:
```
Comment (kazet) [inline artemis/modules/nuclei.py:155]: this is a separate topic, please keep the PRs separate

Comment (kshitijk4poor) [reply artemis/modules/nuclei.py:155]: yep sorry, the fix leaked to the main branch

**Pull request closed without merging**